gpytorch/models/exact_prediction_strategies.py

```diff
@@ -228,6 +239,8 @@ def get_fantasy_strategy(self, inputs, targets, full_inputs, full_targets, full_
                 new_covar_cache = torch.cholesky_solve(new_root.transpose(-2, -1), torch.cholesky(cap_mat))
         new_covar_cache = new_covar_cache.transpose(-2, -1)
 
+        full_targets = full_targets.view(*batch_shape, -1)
```

Comment:
In the case of fast fantasies at repeated inputs, `full_targets` has an extra batch dimension, which is lost here.  Making this `full_targets = full_targets.view(*target_batch_shape, -1)` will fix this issue. The length of `target_batch_shape` can be inferred based on whether `full_output` is a `MultitaskMultivariateNormal`. L256 also needs to be updated.